caffe2/ideep/operators/operator_fallback_ideep.h

```diff
@@ -99,42 +101,48 @@ class IDEEPFallbackOp final : public IDEEPOperator {
     }
 
     for (int i = 0; i < OutputSize(); ++i) {
-      if (SkipOutputCopy::Contains(i)) {
-        VLOG(1) << "Copy output: index " << i << " skipped.";
-        continue;
-      }
       CAFFE_ENFORCE(
           local_output_blobs_[i]->template IsType<TensorCPU>(),
           "IDEEP fallback op currently does not support non-TensorCPU "
           "output type who needs copying.");
-      const auto& src = local_output_blobs_[i]->template Get<TensorCPU>();
+      const auto &src = local_output_blobs_[i]->template Get<TensorCPU>();
 
-      if (src.template IsType<float>()) {
-        Blob* dst = OperatorBase::OutputBlob(i);
-        if (!dst->template IsType<itensor>()) {
+      if (src.template IsType<float>() && !SkipOutputCopy::Contains(i) &&
+          base_op_->type() != "Python") {
+        Blob *dst = OperatorBase::OutputBlob(i);
+        // The output tensor must be ideep tensor with public format.
+        // If reusing ideep tensor with non-public format, the tensor buffer
+        // will be wrong interpreted.
+        if (!dst->template IsType<itensor>() ||
+            !dst->template Get<itensor>().is_public_format()) {
           dst->Reset(new itensor());
         }
```

Comment:
What's the `else` clause? We just go ahead assume the blob contains `itensor` if the condition is not met? I think this condition should be lifted out.

Reply:
if it's not an itensor, the tensor will not be reused. The GetMutable<itensor> in line \#123 will re-create a new itensor for us.